### ISSUE TITLE
inheritdoc tooltips do not display typeparamref correctly

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#64965]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/inheritdoc-tooltips-do-not-display-typep/10173710)._

---
[severity:It's more difficult to complete my work]
When using `` to document a member, if the source documentation contains a `typeparamref` tag, then the tooltip renders the documentation incorrectly.

There are many ways of reproducing this bug, and I've [documented the ones I've found in a gist](https://gist.github.com/bfriesen/d6144163d2697c1e1fa79c2e13ad2ca6). That said, the simplest way of reproducing the bug is to create these files:

```
/// <summary>
/// The generic argument is .
/// </summary>
public class Example
{
}

/// 
public class Inherited : Example
{
}
```

When you view the tooltip documentation for `Example`, it shows "The generic argument is int.":

![image.png](https://aka.ms/dc/image?name=Bae9c914f29cc4f64a288ce0f76e5b839638013731157509499_20221014-153829-image.png&tid=ae9c914f29cc4f64a288ce0f76e5b839638013731157509499)

However, when you view the tooltip documentation for `Inherited`, it shows "The generic argument is TArgument.":

![image.png](https://aka.ms/dc/image?name=B8ec982c898c14b01b36be2a3be86b8ce638013731754414581_20221014-153935-image.png&tid=8ec982c898c14b01b36be2a3be86b8ce638013731754414581)

---

I only recently discovered that it is possible to specify a `cref` with ``, allowing it to be used outside of inheritance situations, meaning I could point a method at any other method. This was great news, because I have a ton of duplicated doc comments in my libraries. So I went through my libraries and eliminated a _ton_ of duplicate doc comments. It was only after a bunch of other commits that I discovered that `` tags weren't being rendered in documentation tooltips. Now I'm going to have to revert a bunch of doc comments (but not all, because not all use `` tags).

---

Out of curiosity: has `` always had the ability to specify a `cref`? If not, when was that functionality added?

---
#### Feedback Bot on 10/16/2022, 09:35 PM: 

(private comment, text removed)
#### Feedback Bot on 10/24/2022, 04:56 PM: 

(private comment, text removed)

---
### Original Solutions
(no solutions)

**Comment by sharwell** [comment]
> has `<inheritdoc/>` always had the ability to specify a `cref`?

Yes, this functionality was added at the same time `<inheritdoc/>` support was added.

**Comment by sharwell** [comment]
> However, when you view the tooltip documentation for `Inherited<int>`, it shows "The generic argument is TArgument.":

Note that there are two different `TArgument` values present. It might be easier to see how type substitution is currently playing into documentation inheritance with this code example:

```csharp
/// <summary>
/// The generic argument is <typeparamref name="TArgument"/>.
/// </summary>
public class Example<TArgument>
{
}

/// <inheritdoc/>
public class Inherited<TArgument2> : Example<TArgument2>
{
}
```

Here the documentation will render as "The generic argument is TArgument2.", which shows that one level of type substitution is occurring but the second is not.
